### PR TITLE
Delegations on user profile

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #780 
Fixes: #697

### CODE REVIEW

**Comment by andorp** [comment]
The school and class level delegations can be buggy, as it is based on the `DelegationListsMap` information. We have to validate if that is the case.

File: src/Frontend/Page/User.hs [hunk 1]
```diff
@@ -310,29 +310,29 @@ createdIdeas userId ideasQuery = do
...
+            ds)
 
 
--- ** User Profile: User As Delegatee
+-- ** User Profile: User As Delegate
 
 instance ToHtml PageUserProfileUserAsDelegate where
     toHtmlRaw = toHtml
-    toHtml p@(PageUserProfileUserAsDelegate ctx (DeletedUser user) _delegations) = semanticDiv p $ do
+    toHtml p@(PageUserProfileUserAsDelegate ctx (DeletedUser user) _delegationListsMap _delegations) = semanticDiv p $ do
```
Comment (fisx) [inline src/Frontend/Page/User.hs]: please rename to `_delegateeListsMap` (or replace with `_`).  or rename the type.

Comment (fisx) [inline src/Frontend/Page/User.hs]: renaming the type is best.

File: src/Frontend/Page/User.hs [hunk 2]
```diff
@@ -334,9 +334,9 @@ instance ToHtml PageUserProfileUserAsDelegate where
                 div_ [class_ "container-narrow"] $ do
                     renderDelegations True delegationListsMap
 
-delegatedVotes :: (ActionPersist m, ActionUserHandler m)
+userProfileUserAsDelegate :: (ActionPersist m, ActionUserHandler m)
```
Comment (fisx) [inline src/Frontend/Page/User.hs]: :+1:

**Comment by fisx** [comment]
LGTM, with the above few nit-picks.

Can be merged when green.

**Comment by andorp** [comment]
LGTM

**Pull request merged**